elements/howto-label/howto-label.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function() {
+  /**
+   * `howtoLabelCounter` counts the number of `<howto-label>` instances created.
+   * The number is used to generated new, unique IDs.
+   */
+  let howtoLabelCounter = 0;
+
+  /**
+   * Cloning contents from a &lt;template&gt; element is more performant
+   * than using innerHTML because it avoids addtional HTML parse costs.
+   */
+  const template = document.createElement('template');
+  template.innerHTML = `
+    <slot></slot>
+  `;
+
+  class HowToLabel extends HTMLElement {
+    static get observedAttributes() {
+      return ['for'];
+    }
+
+    constructor() {
+      super();
+      this.attachShadow({mode: 'open'});
+      this.shadowRoot.appendChild(template.content.cloneNode(true));
+      // Listen for the slotchange event to label any wrapped children
+      this.shadowRoot.querySelector('slot')
+        .addEventListener('slotchange', this._forChanged.bind(this));
```

Comment:
It’s a bit weird to call `_forChanged` on `slotchange`. Maybe you can find a better name?